pkg/module/sync/runner_impl.go

```diff
@@ -224,17 +225,17 @@ func stateFromManifest(obj client.Object) shared.State {
 	}
 }
 
-func DeleteNoLongerExistingModuleStatus(
-	ctx context.Context,
-	kyma *v1beta2.Kyma,
-	moduleFunc GetModuleFunc,
+func DeleteNoLongerExistingModuleStatus(ctx context.Context, kyma *v1beta2.Kyma, moduleFunc GetModuleFunc,
+	removeModuleStateMetrics func(kymaName, moduleName string),
 ) {
 	moduleStatusMap := kyma.GetModuleStatusMap()
 	moduleStatus := kyma.GetNoLongerExistingModuleStatus()
 	for idx := range moduleStatus {
 		moduleStatus := moduleStatus[idx]
 		if moduleStatus.Manifest == nil {
-			metrics.RemoveModuleStateMetrics(kyma, moduleStatus.Name)
+			if removeModuleStateMetrics != nil {
```

Comment:
Also, we could start using guard clauses to be honest. If `removeModuleStateMetrics` is nil, there is not repercussion, so it seems to be fine to use this without it. But actually I would say it's not really optional, we expect the function to be called in this scenario where the `moduleStatus.Manifest == nil`. And if `removeModuleStateMetrics` is nil, it's actually a developer error and we want to know about it and fail fast if that's the case.
The usage of `DeleteNoLongerExistingModuleStatus` in the test file also indicates that this is designed like this to make usage there more "comfortable", because we can just pass nil. But we can instead pass a mock implementation of the interface and can even verify if it's called or not.
WDYT?